docs/guides/web-api-basics.md

```diff
@@ -0,0 +1,444 @@
+
+|Key|Possible Value|
+|-|-|
+|The Web API executor|Any string - the default is `DEFAULT_SINGLETON_EXECUTOR`|
+|Workspace ID|**team_id** (e.g., `T1234567`)|
+|`all_completed_calls`|The numbers of all requests per API method|
+|`successful_calls`|The numbers of all requests that resulted in `"ok": true` responses per API method|
+|`unsuccessful_calls`|The numbers of all requests that resulted in `"ok": false` responses per API method|
+|`failed_calls`|The numbers of all requests that resulted in failures (either of **SlackApiException** or **IOException**) per API method|
+|`current_queue_size`|The sizes of the current queues per API method|
+|`last_minute_requests`|The numbers of all requests in the last 60 seconds per API method|
+|`rateLimitedMethods`|Rate limited method names and their epoch time (ms) suggested by Slack API (according to the `retry-after` response header)|
+
+```js
+{
+  "DEFAULT_SINGLETON_EXECUTOR": {
+    "T1234567": {
+      "all_completed_calls": {
+        "chat.postMessage": 120,
+        "users.info": 2,
+        "conversations.members": 2
+      },
+      "successful_calls": {
+        "chat.postMessage": 110,
+        "users.info": 2,
+        "conversations.members": 2
+      },
+      "unsuccessful_calls": {
+        "chat.postMessage": 7
+      },
+      "failed_calls": {
+        "chat.postMessage": 3
+      },
+      "current_queue_size": {
+        "chat.postMessage_C01ABC123": 5,
+        "users.info": 0
+      },
+      "last_minute_requests": {
+        "chat.postMessage_C01ABC123": 100,
+        "chat.postMessage_C03XYZ555": 3,
+        "users.info": 2,
+        "conversations.members": 2
+      },
+      "rateLimitedMethods": {
+        "chat.postMessage_C01ABC123": 1582183395064
+      }
+    }   
+  }
+}
+```
+
+### Metrics Datastore backed by Redis
+
+If your app wants to have a unified datastore to collect all the metrics across the nodes, we recommend having a Redis cluster for it. It's pretty easy to set up the API client configuration. The following is an example to use a Redis server running on the same host.
+
+```java
+import com.slack.api.Slack;
+import com.slack.api.SlackConfig;
+import com.slack.api.methods.metrics.MetricsDatastore;
+import com.slack.api.methods.metrics.impl.RedisMetricsDatastore;
+
+SlackConfig config = new SlackConfig();
+// brew install redis
+// redis-server /usr/local/etc/redis.conf --loglevel verbose
+JedisPool jedis = new JedisPool("localhost");
+config.getMethodsConfig().setMetricsDatastore(new RedisMetricsDatastore("test", jedis));
+
+Slack slack = Slack.getInstance(config);
+```
+
+---
+
+## Real Time Messaging (RTM)
+
+The Real Time Messaging API is a WebSocket-based API that allows you to receive events from Slack in real-time and send messages as users. It’s sometimes referred to just as the “RTM API”.
```

Comment:
Note that RTM API is not recommended: "The RTM API is not recommended unless your app has unique restrictions, like needs to receive events from behind a firewall."